src/sql/base/browser/ui/panel/panel.component.ts

```diff
@@ -79,23 +87,59 @@ export class PanelComponent extends Disposable {
 	@Input() public actions?: Array<Action>;
 	@ContentChildren(TabComponent) private readonly _tabs!: QueryList<TabComponent>;
 	@ViewChild(ScrollableDirective) private scrollable?: ScrollableDirective;
+	@ViewChild('tabList', { read: ElementRef }) private _tabList!: ElementRef;
 
 	@Output() public onTabChange = new EventEmitter<TabComponent>();
 	@Output() public onTabClose = new EventEmitter<TabComponent>();
 
 	private _activeTab?: TabComponent;
 	private _actionbar?: ActionBar;
 	private _mru: TabComponent[] = [];
+	private _showTabText: boolean = true;
 
 	protected AutoScrollbarVisibility = ScrollbarVisibility.Auto; // used by angular template
 	protected HiddenScrollbarVisibility = ScrollbarVisibility.Hidden; // used by angular template
 	protected NavigationBarLayout = NavigationBarLayout; // used by angular template
 
 	@ViewChild('panelActionbar', { read: ElementRef }) private _actionbarRef!: ElementRef;
-	constructor(@Inject(forwardRef(() => NgZone)) private _zone: NgZone) {
+	constructor(
+		@Inject(forwardRef(() => NgZone)) private _zone: NgZone,
+		@Inject(forwardRef(() => ChangeDetectorRef)) private _cd: ChangeDetectorRef) {
 		super();
 	}
 
+	public get toggleTabLabelButtonClass(): string {
+		return this._showTabText ? 'tab-action collapse' : 'tab-action expand';
+	}
+
+	public get toggleTabLabelButtonLabel(): string {
+		return this._showTabText ? nls.localize('hideTextLabel', "Hide text labels") : nls.localize('showTextLabel', "Show text labels");
+	}
+
+	public get toggleTabLabelButtonExpanded(): boolean {
+		return this._showTabText;
+	}
+
+	updateToggleState(): void {
```

Comment:
>updateToggleState [](start = 1, length = 17)

Just have this be a setter for the _showTabText/collapsed property. This isn't a great name to begin with since it doesn't imply anything about toggling the state (which is what it's actually doing)

Reply:
updated to toggleTabPanel()